src/sentry/sentry_metrics/indexer/strings.py

```diff
@@ -236,6 +238,13 @@ def resolve(self, use_case_id: UseCaseID, org_id: int, string: str) -> Optional[
     def reverse_resolve(self, use_case_id: UseCaseID, org_id: int, id: int) -> Optional[str]:
         if id in REVERSE_SHARED_STRINGS:
             return REVERSE_SHARED_STRINGS[id]
+
+        # HACK: if a string gets re-indexed we need to have some way to look
+        # up the old id and we do it this way because the table has a unique
+        # constraint on the org_id and the string.
+        reindexed_ints = settings.SENTRY_METRICS_INDEXER_REINDEXED_INTS
+        if id in reindexed_ints:
+            return reindexed_ints[id]
```

Comment:
Would be better to have this code execute only if `indexer.reverse_resolve` fails. Even if we forget to remove this code after 90 days the code would still be valid and doing the correct sequence of lookups when the ids from postgres and hardcoded ints overlap.

Reply:
updated